src/cliffordsynthesis/encoding/GateEncoder.cpp

```diff
@@ -135,12 +136,12 @@ void GateEncoder::assertZConstraints(const std::size_t pos,
                                      const std::size_t qubit) {
   const auto& gatesToZTransformations = [this](const auto& p1, const auto& p2,
                                                const auto& p3) {
-    return tvars->singleQubitZChange(p1, p2, p3);
+    return tvars->singleQubitZChange(p1 + 1, p2, p3);
```

Comment:
Very briefly, for my understanding, these offsets are due to the additional "permutation tableau", right?

Reply:
That's right, I also tried implementing it without offsets and then forcing that no Gates are applied in the first and last timestep. But then unnecessary variables for gates are created and need to be set to False, so I chose to use offsets way.

Reply:
Gotcha, given that there is some other encoding techniques that need to be worked into QMAP we just leave that for a future refactoring